### PR TITLE
Tm 6425 export button not working api

### PR DESCRIPTION
**Pull request by fimbriano**
Dual Merge: 
- [FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/3087)

[Ticket](https://metaphase.atlassian.net/browse/TM-6425)

Updated version
Added error handling for client export

### CODE REVIEW

File: talentmap_api/fsbid/services/client.py [hunk 1]
```diff
@@ -600,10 +602,19 @@ def fsbid_assignments_to_tmap(assignments):
 
 
 def fsbid_languages_to_tmap(languages):
+    # if no languages present (languages: [])
+    if languages is None:
+        return []
+
```
Comment (M00staff) [inline talentmap_api/fsbid/services/client.py:608]: 👍

File: talentmap_api/fsbid/services/client.py [hunk 2]
```diff
@@ -600,10 +602,19 @@ def fsbid_assignments_to_tmap(assignments):
...
     tmap_languages = []
     empty_score = '--'
     for x in languages:
-        if not x.get('empl_language', None) or not str(x.get('empl_language')).strip():
+        # if x is None or not a dict, skip
+        if x is None or not isinstance(x, dict):
+            if x is None:
+                logger.warning(f"Skipping None value in languages: {languages}\n")
+            continue
+        if not x.get('empl_language', None) or not str(x.get('empl_language', None)).strip():
```
Comment (M00staff) [inline talentmap_api/fsbid/services/client.py:617]: 👍

File: talentmap_api/fsbid/services/client.py [hunk 3]
```diff
@@ -613,19 +624,35 @@ def fsbid_languages_to_tmap(languages):
             "test_date": ensure_date(x.get('empl_high_test_date', None)),
             "speaking_score": s or empty_score,
             "reading_score": r or empty_score,
-            "custom_description": f"{str(x.get('empl_language_code')).strip()} {s or empty_score}/{r or empty_score}"
+            "custom_description": f"{str(x.get('empl_language_code', None)).strip()} {s or empty_score}/{r or empty_score}"
```
Comment (M00staff) [inline talentmap_api/fsbid/services/client.py:627]: TIL: running `str(None)` will return the string "None" ... neat!

File: talentmap_api/fsbid/services/client.py [hunk 4]
```diff
@@ -613,19 +624,35 @@ def fsbid_languages_to_tmap(languages):
...
+    if not languages:
+        return "None"
+    
     tmap_language_only = []
     for x in languages:
-        if not x.get('empl_language', None) or not str(x.get('empl_language')).strip():
+        # checks for non-dict elements such as [numbers, strings, None, etc] or
+        # if the "empl_language" key is not present, skip
+        if not isinstance(x, dict) or "empl_language" not in x:
+            logger.warning(f"Skipping invalid language: {x}\n")
```
Comment (M00staff) [inline talentmap_api/fsbid/services/client.py:642]: 👍

File: talentmap_api/fsbid/services/client.py [hunk 5]
```diff
@@ -613,19 +624,35 @@ def fsbid_languages_to_tmap(languages):
...
             continue
 
-        tmap_language_only.append(
-            str(x.get('empl_language')).strip() if x.get('empl_language') else x.get('empl_language') or None,
-        )
+        # checks if the value for empl_language key is not a string, skip
+        if not isinstance(x.get('empl_language'), str):
```
Comment (M00staff) [inline talentmap_api/fsbid/services/client.py:646]: I'm curious what kind of data were we getting back here that was causing issues?

File: talentmap_api/fsbid/services/client.py [hunk 6]
```diff
@@ -613,19 +624,35 @@ def fsbid_languages_to_tmap(languages):
...
+            continue
+
+        # get the value for empl_language key, if its not present, give back None
+        # strip the value to remove any leading/trailing whitespace
+        empl_language = str(x.get('empl_language', None)).strip()
+        if not empl_language:
+            continue
+        
+        tmap_language_only.append(empl_language.strip())
```
Comment (M00staff) [inline talentmap_api/fsbid/services/client.py:655]: 👍

Great comments - thank you for those!

**Comment by M00staff** [comment]
this checks out - we have A LOT of issues with the types of data we get back from FSBID, and their subsequent handling ... there is a whole mess we had to resolve with saving dates a while back where you'll see some logic handling the 'Z' in timestamps 

I'm curious what the data coming back looks like though 🤔 like the outliers specifically that we're handling, here

**Review by M00staff — APPROVED** [review]

File: talentmap_api/fsbid/services/client.py (hunk 5 above)
Comment (fimbriano) [reply talentmap_api/fsbid/services/client.py:646]: `{ null, null }` which then converts to `{ None, None }` when we call `.json` on the response

**Pull request merged**